### DISCUSSION TITLE
how to get selected value of OptionList

### DISCUSSION THREAD

**Discussion by MohamedElgamal** [Textualize/textual#3247] [Q&A] [answered]
i am trying to get the selected value of OptionList and can't figure out which attribute to get the value 
<pre>
 @on(OptionList.OptionMessage)
    def user_selected(self, event: OptionList.OptionMessage):
        print(f"Option Selected {event}")
</pre>
this is my code snippet to catch the selected value

**Comment by willmcgugan** [comment]
Does this help? https://textual.textualize.io/widgets/option_list/#textual.widgets._option_list.OptionList.OptionMessage

**Reply by MohamedElgamal** [reply]
yes, now i have the object i want the value itself, i tried to figure out the attribute
<pre>
    @on(OptionList.OptionMessage)
    def user_selected(self, event: OptionList.OptionSelected):
        option_id = event.option_index
        option = event.option_list
        value = option.get_option_at_index(option_id)
        print(f"Option Selected {value}")
    </pre>

**Reply by davep** [reply, marked as answer]
`Option`s within an `OptionList` don't have a value. The idea behind `OptionList` is that it's a list of prompts (options) that the user can select from, and when you have the index of a selection option you can decide to do something with that. For example, you maybe maintaining a parallel list of values and the index can be used to retrieve the value from that.

If you wanted to keep a value along with an `Option` in an `OptionList` the thing to do would be to subclass `Option` for your own needs and then add appropriate properties. For example, in a personal library I maintain, I used `OptionList` as the heart of a filesytem entry picking dialog, and [I inherit from `Option` and add lots of properties for holding details about entries in the filesystem](https://github.com/davep/textual-fspicker/blob/6adec86e3a6b089913c999e1c5c5f2a35fc70ae5/textual_fspicker/parts/directory_navigation.py#L52).

**Reply by TomJGooding** [reply]
@MohamedElgamal Assuming you just want to show a list of selectable users like the previous discussion #3238, perhaps a [SelectionList](https://textual.textualize.io/widgets/selection_list/) would be more suitable?

**Reply by MohamedElgamal** [reply]
@TomJGooding only one user should be selected at a time i was going to use select but there is no option to write text and autocomplete also tried to use <a href= "https://github.com/darrenburns/textual-autocomplete">textual-autocomlete</a> but it does't support mouse so i am think to do as follow user will type in input field and with every key strick Option List will display the values very close to this value to select one of it
